### PR TITLE
[MAT-2758] Split up New Measure Name options into separate checkboxes

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
**[MAT-2758](https://jira.cms.gov/browse/MAT-2758)**

Split the CQL Library Name option checkbox into two and disable the textbox whenever
the user elects to have the name automatically generated to match the generated
CMS ID.

## Description
Splitting the option into its components makes it clearer to the user what affect each option will have on the resulting measure. These checkboxes should only appear on **New** measure and composite measure pages.

Users may elect to generate a CMS/eCQM ID for the measure and, when selected, they may also elect to match the related CQL Library Name with the generated ID. When a user elects to have both the ID and Library Name generated, the Library Name's textbox is disabled and all field level validations are cleared.

The CMS ID and Library Name are generated on **Save and Continue**. This greatly reduces the chances of another user sniping the ID before the save operation can complete as well as helping to ensure the ID's value increases in numerical order. Also, since the Measure and Library name validators are abstracted away from the presenter, it was difficult to remove the Library name validation whenever the field was disabled. Generating the ID and Name at the start of the save  operation allows for the validation to run normally, which keeps our Library name template in line with what we require of our users.

### Other goodies:
- Modify the Cypress defaults to retain cookies between tests, otherwise MAT fails to successfully run more than one test case
- Correctly ignore mat-cypress bulk in gitignore
- Simplify getValidations implementation
- Condense handler logic and clear validation errors when TextArea is disabled

## JIRA Ticket
[MAT-2758](https://jira.cms.gov/browse/MAT-2758): Split up option to Generate CMS ID for new measures

## Checklist

- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases.
- [x] Tests have been run locally and pass.
Screen Shot 2021-03-20 at 5 35 14 PM

### **Single checkbox split into two:**
Screen Shot 2021-03-08 at 6 06 35 PM
### **Second checkbox enabled when first is checked:**
Screen Shot 2021-03-08 at 6 06 41 PM
### **Related TextBox is disabled when second checkbox is checked:**
Screen Shot 2021-03-08 at 6 07 08 PM
### **Above measure's Generate CMS ID option successful:**
Screen Shot 2021-03-08 at 6 08 29 PM
### **Selecting both options:**
Screen Shot 2021-03-09 at 4 36 50 PM
### **Generate CMS ID successful:**
Screen Shot 2021-03-09 at 4 37 16 PM
### **Match CQL Library Name successful:**
Screen Shot 2021-03-09 at 4 37 37 PM
### **Draft and Clone pages hide checkboxes:**
Screen Shot 2021-03-10 at 2 41 36 PM
Screen Shot 2021-03-10 at 2 39 12 PM
### **New Composite Measure:**
Screen Shot 2021-03-10 at 6 35 23 PM

### CODE REVIEW

File: src/main/java/mat/client/measure/AbstractNewMeasureView.java [hunk 1]
```diff
@@ -323,24 +338,37 @@ protected void addCompositeMeasureModelType() {
...
+        generateCmsIdCheckbox.setTitle("Click to generate an eCQM ID on Save.");
         generateCmsIdCheckbox.setId("generateCmsId_CheckBox");
         ((Element) generateCmsIdCheckbox.getElement().getChild(0)).setAttribute("for", "generateCmsId_Input");
         ((Element) generateCmsIdCheckbox.getElement().getChild(0).getChild(0)).setAttribute("id", "generateCmsId_Input");
-        generateCmsIdGroup.add(generateCmsIdCheckbox);
+        nameAndIdOptionGroup.add(generateCmsIdCheckbox);
     }
 
+    protected void addMatchLibraryNameToCmsIdCheckbox() {
+        // TODO Add cy id for cypress.io testing
```
Comment (RohitKandimalla) [inline src/main/java/mat/client/measure/AbstractNewMeasureView.java:360]: Todo can be removed?

**Review by RohitKandimalla — DISMISSED** [review]

File: mat-cypress/cypress/integration/measures/NewMeasureOptions.spec.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    cy.get(measureLibrary.newMeasureButton).click();
+    cy.wait(1000);
+
+    cy.get(newMeasurePage.generateCmsIdCheckbox).should('be.enabled');
+    cy.get(newMeasurePage.matchLibraryNameCheckbox).should('be.disabled');
+    cy.get(newMeasurePage.cqlLibraryName).should('be.enabled');
+  });
+});
+
+// describe('New Measure: Generate CMS/eCQM ID', () => {
```
Comment (jkotanchik-SB) [inline mat-cypress/cypress/integration/measures/NewMeasureOptions.spec.js:98]: This was intentionally commented out. It needs to be moved to its own test suite since its testing beyond the New Measure UI.

Comment (jkotanchik-SB) [reply mat-cypress/cypress/integration/measures/NewMeasureOptions.spec.js]: Removed since it wasn't going to be completed in this PR.

**Review by RohitKandimalla — APPROVED** [review]
